types.ts

```diff
@@ -0,0 +1,16 @@
+export interface Storage<T> {
+  put(record: T): void;
+  get(start?: number, end?: number): T[];
+}
+
+export interface Worker<T, U> {
+  work(records: T[]): Promise<U[]>;
+}
+
+export interface BufIOConfig<T, U> {
+  storage: Storage<T>;
+  worker: Worker<T, U>;
+  batchSize?: number;
+  flushInterval?: number;
+  onError?: (err: Error, records: T[]) => void;
+}
```

Comment:
This in src/io.ts and might be better to make them all classes instead of interfaces